frontend/test/metabase/scenarios/visualizations/pie_chart.cy.spec.js

```diff
@@ -0,0 +1,52 @@
+import { restore, visitQuestionAdhoc } from "__support__/cypress";
+import { SAMPLE_DATASET } from "__support__/cypress_sample_dataset";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATASET;
+
+const testQuery = {
+  type: "query",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+    breakout: [["field-id", PRODUCTS.CATEGORY]],
+  },
+  database: 1,
+};
+
+describe("scenarios > visualizations > pie chart", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  function ensurePieChartRendered(rows, totalValue) {
+    cy.get(".Visualization").within(() => {
+      // detail
+      cy.findByText("Total").should("be.visible");
```

Comment:
This is a good place to talk about explicit vs implicit assertions.
In general, 99% of the time just using `cy.findByText(string)` is more than enough. It already makes sure something exists (implicit assertion).

In very very rare occasions we need to make sure an element is visible before doing something with it. The best example is when we need to click on something. `cy.click()`is going to fail if an element is not visible so asserting and basically waiting for it to become visible explicitly makes sense for such occasions.

Unless there was a specific reason for this assertion, this line would work the same with just `cy.findByText("Total")`. Hope that makes sense and it helps.

Reply:
I see one potential issue with implicit assertions that can cause bugs. For example, in this particular case, the element with the `total` value was present in DOM but it is just hidden. If we had a test that only checks `cy.findByText("Total")` it would miss this bug. Having fully explicit checks is more verbose which might affect readability but it is more reliable. What do you think about this case?

Reply:
If that was the case here in particular then by all means - yeah, leave it as an explicit assertion please. I'm just saying that we need it very rarely and this was probably one of those rare cases.

Reply:
Yes, we may omit these when we deal with something that does not have any changing visibility logic. Thanks for bringing it up